src/arch/x86/x86_64/src/bootblock.S

```diff
@@ -184,7 +189,7 @@ gdtptr16:
 
 	movl	%cr0, %eax
 	andl	$0x7FFAFFD1, %eax /* PG,AM,WP,NE,TS,EM,MP = 0 */
-	orl	$(CD | NW | PE), %eax
+	orl	$({cd} | {nw} | {pe}), %eax
```

Comment:
About the names inside the asm environment, do those have to be lower case (or is it convention to do that)?

I'd rather not have a constant be known under multiple different names, if it can be avoided.

Instead of `{cd}`, could it just be `{X86_CR0_CD}` like the rust constant is named anyway?

Reply:
or even have the crate name in there somehow? I have no idea how this works.

Reply:
global_asm It works kinda like (named) parameters in format strings (see .net, Python).

So you can have: `"hello {foo}, I heard {foo}".format(foo=2)` => `"hello 2, I heard 2"`. I don't think that having a `::` in the name of the placeholder is possible (would conflict with actual formatting specifier)--but if so, that would be nice indeed. (what is possible is to further enter into structs etc)

The assembly that will eventually become stable https://github.com/Amanieu/rfcs/blob/inline-asm/text/0000-inline-asm.md also supports something similar to this pull request in this regard.